src/xpcc/architecture/platform/driver/i2c/stm32l4/i2c_master.cpp.in

```diff
@@ -124,6 +136,58 @@ namespace
 	}
 
 	static inline void
+	callReadOperation(const bool startCondition)
+	{
+		DEBUG_STREAM("read op: reading=" << reading.length);
+		DEBUG_STREAM("nextOperation=" << nextOperation);
+
+		// Only 255 bytes can be written at once.
+		bool autoend = false;
+		bool reload = false;
+		uint8_t nbytes = (reading.length > 255) ? 255 : reading.length;
+
+		if ((nextOperation != xpcc::I2c::Operation::Stop) or (reading.length > 255)) {
+			// RELOAD if the operation continues
+			reload = true;
+		} else if ((nextOperation == xpcc::I2c::Operation::Stop)) {
+			// AUTOEND has no effect if RELOAD is set.
+			autoend = true;
+		}
+
+		DEBUG_STREAM("autoend=" << autoend << ", reload=" << reload << ", nbytes=" << nbytes);
+
+		I2C{{ id }}->CR2 = (autoend ? I2C_CR2_AUTOEND : 0) |\
```

Comment:
Same here with `|\`.

Reply:
More is always better, you know! Just fixed.